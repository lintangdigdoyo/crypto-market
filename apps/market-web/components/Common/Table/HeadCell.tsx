import classNames from "classnames";

interface HeadCellProps extends React.ThHTMLAttributes<HTMLTableCellElement> {
  className?: string;
  children: React.ReactNode;
}

const HeadCell = ({ className, children, ...restProps }: HeadCellProps) => {
  return (
    <th
      className={classNames("p-5 text-gray font-semibold", className)}
      {...restProps}
    >
      {children}
    </th>
  );
};

export default HeadCell;
