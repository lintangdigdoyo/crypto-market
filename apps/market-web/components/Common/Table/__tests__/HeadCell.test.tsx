import renderer from "react-test-renderer";
import "@testing-library/jest-dom";

import HeadCell from "../HeadCell";

describe("HeadCell components", () => {
  it("should renders HeadCell correctly", () => {
    const props = {
      children: <>children</>,
    };
    const component = renderer.create(<HeadCell {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
