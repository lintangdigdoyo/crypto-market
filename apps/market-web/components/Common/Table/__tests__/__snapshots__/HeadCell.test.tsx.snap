// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadCell components should renders HeadCell correctly 1`] = `
<th
  className="p-5 text-gray font-semibold"
>
  children
</th>
`;
